airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/pipeline.py

```diff
@@ -54,7 +54,9 @@ async def run_connector_test_pipeline(context: ConnectorTestContext, semaphore:
 
     all_steps_to_run += get_test_steps(context)
 
-    if not context.code_tests_only:
+    # Note: Static analysis will still run when both metadata.yaml and test files are modified together,
+    # as this is an unlikely scenario and we prefer to err on the side of running the checks.
+    if not context.code_tests_only and not (context.metadata_yaml_only_change or context.test_only_change):
```

Comment:
actually, get rid of metadata_yaml_only_change entirely and delete the property from context.py. We should run these checks whenever metadata.yaml has been modified